code/game/machinery/spaceheater.dm

```diff
@@ -176,6 +177,15 @@
 		SStgui.update_uis(src)
 		return TRUE
 
+/obj/machinery/space_heater/attack_hand_secondary(mob/user, list/modifiers)
+	if(!panel_open)
+		if(!can_interact(user))
+			return
+		on = !on
```

Comment:
Please refactor the action from `ui_act` into a separate proc and call it here.

Reply:
I actually had to remove the `if(!panel_open)` since the space heater is a machine that can be turned ON/OFF when it's open and has some extra features around that.
Still if there is anything on the new version that should be changed...